streams-tests/shared/src/test/scala/zio/stream/ZStreamSpec.scala

```diff
@@ -2804,18 +2804,30 @@ object ZStreamSpec extends ZIOBaseSpec {
             } yield assert(res)(equalTo(List(1, 2, 3, 4)))
           )
         ),
-        testM("repeatEffectWith")(
-          for {
-            ref <- Ref.make[List[Int]](Nil)
-            fiber <- ZStream
-                      .repeatEffectWith(ref.update(1 :: _), Schedule.spaced(10.millis))
-                      .take(2)
-                      .runDrain
-                      .fork
-            _      <- TestClock.adjust(50.millis)
-            _      <- fiber.join
-            result <- ref.get
-          } yield assert(result)(equalTo(List(1, 1)))
+        suite("repeatEffectWith")(
+          testM("succeed")(
+            for {
+              ref <- Ref.make[List[Int]](Nil)
+              fiber <- ZStream
+                        .repeatEffectWith(ref.update(1 :: _), Schedule.spaced(10.millis))
+                        .take(2)
+                        .runDrain
+                        .fork
+              _      <- TestClock.adjust(50.millis)
+              _      <- fiber.join
+              result <- ref.get
+            } yield assert(result)(equalTo(List(1, 1)))
+          ),
+          testM("allow schedule rely on effect value")(checkM(Gen.int(1, 100)) { (length: Int) =>
+            var x        = length
+            val effect   = UIO(if (x <= 0) 0 else { val y = x; x -= 1; y })
+            val schedule = Schedule.identity.addDelay((i: Int) => (if (i == 0) 1 else 0).seconds).forever
+            for {
```

Comment:
As far as I can see, this test doesn't check timings despite it uses Clock.
I feel that we can achieve the same coverage with a simpler testcase.
What about, e.g.:
1) take an effect that counts down from 100 to 0, but crashes on 51
2) pick a schedule that stops when it sees `50`
3) check that it collected (100..50)
4) optionally the schedule might as well crash on elements below 49

Reply:
@simpadjo let me give that a try

Reply:
@simpadjo thanks for the suggestions, done.